diff --git a/onnxruntime/core/framework/tensorprotoutils.cc b/onnxruntime/core/framework/tensorprotoutils.cc
index 6af78f18fb..f32d3253e6 100644
--- a/onnxruntime/core/framework/tensorprotoutils.cc
+++ b/onnxruntime/core/framework/tensorprotoutils.cc
@@ -956,24 +956,38 @@ Status GetExtDataFromTensorProto(const Env& env, const ORTCHAR_T* model_path,
                                  if (fileName.startsWith('./')) {
                                    fileName = fileName.substring(2);
                                  }
-                                 const fileData = Module.MountedFiles.get(fileName);
-                                 if (!fileData) {
-                                   return 2;  // File not found in preloaded files.
-                                 }
-                                 const offset = $1 >>> 0;
-                                 const length = $2 >>> 0;
-                                 const buffer = $3 >>> 0;
-
-                                 if (offset + length > fileData.byteLength) {
-                                   return 3;  // Out of bounds.
-                                 }
 
-                                 try {
-                                   // Copy the file data (fileData,offset,length) into WebAssembly memory (HEAPU8,buffer,length).
-                                   HEAPU8.set(fileData.subarray(offset, offset + length), buffer);
-                                   return 0;
-                                 } catch {
-                                   return 4;
+                                 if (Module.MountedFiles.get('fakefakefake_' + fileName)) {
+                                   const length = $2 >>> 0;
+                                   const buffer = $3 >>> 0;
+                                   try {
+                                     // Set fake zero data to buffer.
+                                     const dummyData = new Uint8Array(length);
+                                     HEAPU8.set(dummyData, buffer);
+                                     return 0;
+                                   } catch {
+                                     return 4;
+                                   }
+                                 } else {
+                                   const fileData = Module.MountedFiles.get(fileName);
+                                   if (!fileData) {
+                                     return 2;  // File not found in preloaded files.
+                                   }
+                                   const offset = $1 >>> 0;
+                                   const length = $2 >>> 0;
+                                   const buffer = $3 >>> 0;
+
+                                   if (offset + length > fileData.byteLength) {
+                                     return 3;  // Out of bounds.
+                                   }
+
+                                   try {
+                                     // Copy the file data (fileData,offset,length) into WebAssembly memory (HEAPU8,buffer,length).
+                                     HEAPU8.set(fileData.subarray(offset, offset + length), buffer);
+                                     return 0;
+                                   } catch {
+                                     return 4;
+                                   }
                                  }
                                }),
                                external_data_file_path.c_str(),
